packages/synapse-react-client/src/components/SynapseTable/SynapseTable.tsx

```diff
@@ -195,6 +198,45 @@ export function SynapseTable(props: SynapseTableProps) {
   })
 
   const { dataHasBeenPrefetched } = usePrefetchTableData()
+  const renderTableDataPlaceholder = !dataHasBeenPrefetched
+
+  const TableCellRenderer = useCallback(
+    (cell: Cell<Row, unknown>) => {
+      const selectColumn = selectColumns.find(cm => cm.name === cell.column.id)
+
+      const shouldWrapInExpandable =
+        selectColumn &&
+        selectColumn.columnType !==
+          ColumnTypeEnum.JSON /* JSON handles its own overflow*/
+      const TableDataCellElement = shouldWrapInExpandable
+        ? ExpandableTableDataCell
+        : 'td'
+
+      return (
+        <TableDataCellElement
+          key={cell.id}
+          style={{
+            width: `calc(var(${getColumnSizeCssVariable(
+              cell.column.id,
+            )}) * 1px)`,
+            textAlign: cell.column.columnDef.meta?.textAlign,
+          }}
+        >
+          {renderTableDataPlaceholder ? (
+            <p>
+              <Skeleton width={'80%'} height={'20px'} />
+            </p>
+          ) : (
+            flexRender(cell.column.columnDef.cell, cell.getContext())
+          )}
+        </TableDataCellElement>
+      )
+    },
+    // This renderer must also update when our `columnVisibility` state changes.
+    // There is some issue with how this updated function component triggers rerendering the React.memoized TableBody component
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [renderTableDataPlaceholder, selectColumns, columnVisibility],
```

Comment:
I spent most of a day trying to figure this out, I may come back to it before I wrap up other changes associated with PORTALS-2970.

One of our tests reliably fails when this breaks, so I think it's pretty safe to leave it like this for now.